thing/index.js

```diff
@@ -477,15 +477,19 @@ function ThingShadowsClient(deviceOptions, thingShadowOptions) {
             discardStale: true,
             enableVersioning: true,
             qos: 0,
-            pending: false
+            pending: false,
+            pendingDeltaSubscribe: true,
+            pendingPersistentSubscribe: true
```

Comment:
I think it's better not to treat delta ad other shadow topic subscriptions separately. One flag should be enough to tell if the shadow is ready to take actions. 

In the same vein, callback should also be invoked only once to be deterministic and avoid user confusion.